File: trunk/java/SR-RBD/src/ddb/communication/DataOutputStream.java
package ddb.communication;

import java.io.IOException;
import java.io.OutputStream;

import ddb.Config;

public class DataOutputStream extends java.io.DataOutputStream
{
	public DataOutputStream(OutputStream out) {
		super(out);
	}
	
	public void writeString(String s) throws IOException {
		
		// transform string to bytes
		byte[] bytes = s.getBytes(Config.getCharset());
		
		// write length
		writeInt(bytes.length);
		
		// write content to stream
		write(bytes, 0, bytes.length);
	}
}

File: trunk/C#/RBD/RBD/Communication/DataOutputStream.cs
// + TODO check

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.IO;
using System.Net;

namespace RBD.Communication
{
    public class DataOutputStream : BinaryWriter
    {
        //public class DataInputStream extends java.io.DataInputStream
        //public DataInputStream(InputStream in) : base {
        public DataOutputStream(Stream output)
            : base(output)
        {
        }

        public void writeBoolean(bool value)
        {
            Write(value);
        }

        public void writeInt(int value)
        {
            Write(IPAddress.HostToNetworkOrder(value));
        }

        public void writeString(String s)
        {
            WriteString(s);
        }

        public void WriteString(String s) //throws IOException
        {
            // transform string to bytes
            byte[] bytes = Config.GetEncoding().GetBytes(s);

            // write length
            writeInt(bytes.Length);

            // write content to stream
            Write(bytes, 0, bytes.Length);
        }
    }

}

